p2p/net/mock/mock_peernet.go

```diff
@@ -160,11 +160,41 @@ func (pn *peernet) connect(p peer.ID) (*conn, error) {
 func (pn *peernet) openConn(r peer.ID, l *link) *conn {
 	lc, rc := l.newConnPair(pn)
 	log.Debugf("%s opening connection to %s", pn.LocalPeer(), lc.RemotePeer())
+	addConnPair(pn, rc.net, lc, rc)
```

Comment:
Shouldn't this also have a rc.net.addConn below?

Reply:
It happens in the other goroutine in `remoteOpenConn`.